### PR TITLE
feat(aws-ecs): Added dockerVolumeConfiguration to Volume

### LINKED ISSUE #1356: DockerVolumeConfiguration support
Since dockerVolumeConfiguration is not supported for Volume, it takes time to use docker volume.

### PR DESCRIPTION
**Pull request by cohalz**
Fixes #1356

----

### Pull Request Checklist

Please check all boxes, including N/A items:

#### Testing

- [x] Unit test and/or integration test added
- [x] __Toolkit change?:__ [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) manually executed (paste output to the PR description)
- [x] __Init template change?:__ coordinated update of integration tests (currently maintained in a private repo).

#### Documentation

- [x] __README__: README and/or documentation topic updated
- [x] __jsdocs__: All public APIs documented

### Title and description

- [x] __Change type__: Title is prefixed with change type:
  * `fix(module): <title>` bug fix (_patch_)
  * `feat(module): <title>` feature/capability (_minor_)
  * `chore(module): <title> ` won't appear in changelog
  * `build(module): <title>` won't appear in changelog
- [x] __Title format__: Title uses lower case and doesn't end with a period
- [x] __Breaking change?__: Last paragraph of description is: `BREAKING CHANGE: <describe exactly what changed and how to achieve similar behavior + link to documentation/gist/issue if more details are required>`
- [x] __References__: Indicate issues fixed via: `Fixes #xxx` or `Closes #xxx`

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts [hunk 1]
```diff
@@ -359,6 +363,29 @@ export interface Host {
   sourcePath?: string;
 }
 
+export interface DockerVolumeConfiguration {
+  /**
+   * If true, the Docker volume is created if it does not already exist
+   */
+  autoprovision?: boolean,
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts]: please use ";" at the end of each prpoerty

File: packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts [hunk 2]
```diff
@@ -359,6 +363,29 @@ export interface Host {
...
+  /**
+   * The Docker volume driver to use
+   */
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts]: Add "@default" with a description to all optional fields

File: packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts [hunk 3]
```diff
@@ -346,7 +346,11 @@ export interface Volume {
    * A name for the volume
    */
   name?: string;
-  // FIXME add dockerVolumeConfiguration
+
+  /**
+   *   Specifies the configuration of a Docker volume
```
Comment (eladb) [inline packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts]: Two spaces at the beginning of the line

**Comment by clareliguori** [comment]
cc @SoManyHs

File: packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts [hunk 4]
```diff
@@ -346,7 +346,11 @@ export interface Volume {
    * A name for the volume
    */
   name?: string;
-  // FIXME add dockerVolumeConfiguration
+
+  /**
+   * Specifies this configuration when using Docker volumes
+   */
+  dockerVolumeConfiguration?: DockerVolumeConfiguration;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts]: I wonder if it makes sense to have these inline in the `Volume` struct.

As well as, why are the name and the configuration optional? Does it make sense to leave them optional?

I know @SoManyHs was working on a higher-level abstraction over volumes, so I'd like her to weigh in as well.

Comment (cohalz) [reply packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-volumes.html
Thanks, I changed `name` to required.

**Pull request merged**